Users can view a particular bet

YarekTenko opened the pull request:
Users can view particular bets by accessing /bets/:id.

LeonTenko committed:
Add belongs_to :creator relationship to Bet

Added belongs_to :creator to Bet and has_many :created_bets to User.

ghost commented:
Minor adjustment: now the bets and users are displayed in for of a table instead of card blocks.

qr8r commented on app/models/bet.rb:
```diff
@@ -2,6 +2,7 @@ class Bet < ApplicationRecord
   has_many :users, through: :user_bets
   has_many :user_bets
   has_many :options, class_name: 'BetOption'
+  belongs_to :creator, class_name: 'User', foreign_key: :creator_id
```
Is the foreign_key declaration required here?

qr8r commented on app/models/user.rb:
```diff
@@ -6,4 +6,8 @@ class User < ApplicationRecord
   has_secure_password
+
+  def waged_on(bet_option)
+    user_bets.where(bet_option: bet_option).pluck(:amount_bet).sum
```
Can a user bet on the same option more than once?

qr8r commented on app/views/bets/show.html.erb:
```diff
@@ -1,49 +1,73 @@
-  <div class="row">
-    <div class="col col-lg-12">
-      <h2>Viewing bet: <%= @bet.id %></h2>
+<!-- Bet -->
```
Write code so good that doesnt need comments ;)

qr8r commented on app/views/bets/show.html.erb:
```diff
@@ -1,49 +1,73 @@
+      </div>
+
+      <div class="card-footer text-danger">
+        Expires on: <%= @bet.expires_at.strftime('%A, %B %d %Y, %l:%M %P') %>
```
:+1: `#strftime`

qr8r commented on app/views/bets/show.html.erb:
```diff
@@ -1,49 +1,73 @@
+    <div class="card">
+      <div class="card-header text-muted">
+        <div class="row">
+          <div class="col col-lg-4">User</div>
```
![divitis](https://cdn.slidesharecdn.com/ss_thumbnails/fight-against-divitis-120730100548-phpapp01-thumbnail-4.jpg?cb=1343643142)

qr8r replied:
We lost the fight to divitis.... but the functionality meets the story

qr8r commented on app/views/bets/show.html.erb:
```diff
@@ -1,49 +1,73 @@
+      </div>
+      <div class="card-block">
+        <table class="table table-hover">
+          <% @bet.user_bets.each do |ub| %>
```
Might be a good opportunity for a `<%= render @bets.user_bets %>`

qr8r reviewed (approved):
Minor comments, but overall :1st_place_medal:

qr8r reviewed (changes requested):
Meant to request changes earlier

jacquesporveau commented on app/views/bets/show.html.erb:
```diff
@@ -1,49 +1,73 @@
+            <tr class="row">
+              <td class="col col-lg-4"><%= i + 1 %></td>
+              <td class="col col-lg-4"><%= option.option_text %></td>
+              <td class="col col-lg-4">$<%= option.option_total %></td>
```
Mmmmmm the naming of the BetOption model, and option_total method are kind of weird. You shouldn't need to specify that you are looking for the "option" total when you are calling it on an object called option. Just `option.total`. 

I realize that this isn't really related to this PR but it's just something to keep in mind.

qr8r replied:
:+1: Naming is hard, but I do agree. We dont need to change this now. But definitely something to keep in mind for the future

qr8r commented:
Jenkins test this please

cbrunsdon commented:
jenkins test this please

LeonTenko committed:
Display Bet creator, text and expiry date

LeonTenko committed:
Display UserBets for a particular bet

-Diplays User, the option they picked and the amount of money they
bet on that option.

LeonTenko committed:
Display Bet options for a particular Bet

-Displays the option number(ordinary, not actual BetOption ID),
option text and the amount of money that was bet on that option.

jacquesporveau commented on app/views/bets/show.html.erb:
```diff
@@ -1,49 +1,16 @@
+  <div class="col col-lg-6 offset-lg-3">
+    <div class="card text-center">
+      <div class="card-header text-muted">
+        <h4><%= @bet.creator.username %></h4>
```
Is creator a required association on bet? What would happen here if a bets creator was `nil`? Same thing with username, I am not seeing any validations on a users username. What if they didn't have a user name?

YarekTenko replied:
Yeah, basically, creator can't be nil, otherwise the Bet won't be created.
Username is not validated yet, but there is a separate story for that, so I decided not to implement it here.
Thanks for the feedback!
